pkg/api/user_token.go

```diff
@@ -62,11 +63,45 @@ func (server *HTTPServer) getUserAuthTokensInternal(c *models.ReqContext, userID
 			isActive = true
 		}
 
+		parser := uaparser.NewFromSaved()
+		client := parser.Parse(token.UserAgent)
+
+		osVersion := ""
+		if client.Os.Major != "" {
+			osVersion = client.Os.Major
+
+			if client.Os.Minor != "" {
+				osVersion = osVersion + "." + client.Os.Minor
+			}
+		}
+
+		browserVersion := ""
+		if client.UserAgent.Major != "" {
+			browserVersion = client.UserAgent.Major
+
+			if client.UserAgent.Minor != "" {
+				browserVersion = browserVersion + "." + client.UserAgent.Minor
+			}
+		}
+
 		result = append(result, &dtos.UserToken{
-			Id:        token.Id,
-			IsActive:  isActive,
-			ClientIp:  token.ClientIp,
-			UserAgent: token.UserAgent,
+			Id:       token.Id,
+			IsActive: isActive,
+			ClientIp: token.ClientIp,
+			UserAgent: map[string]map[string]string{
```

Comment:
I think we can start simple here and skip versions. Based on my earlier experience with this library something like this should probably do it:

```go
Device:          client.Device.ToString(),
OperatingSystem: client.Os.Family,
Browser:         client.UserAgent.Family,
```

Reply:
I did the key/values for flexibility with use. Also, for the mockup in #16223, need the version for OS so felt like it might be worth just making the info available across for whomever. Can move them up to top level but still have their val maps.